### ISSUE TITLE
OpenPopup after BeginPopup fails when no windows are focused

### ISSUE THREAD

**Issue by cfillion** [ocornut/imgui#4308]
**Version/Branch of Dear ImGui:**

Version: 1.83
Branch: master and docking

**My Issue:**

Popups fail to open when `OpenPopup` is called after `BeginPopup` and there are no  focused windows. (EDIT: To be precise, it works as expected either if `OpenPopup` is used before the `BeginPopup`, or if *any* window has focus.)

This appears to be because `NewFrame` calls `ClosePopupsOverWindow` even when `g.NavWindow` is NULL.

https://github.com/ocornut/imgui/blob/5dd1e38b7a5a6ed6c6d94c54c6eb54ba3a88ed81/imgui.cpp#L4105

**Screenshots:**

![Demo > Popups & Modal windows > Context menus](https://i.imgur.com/bpFuqhT.gif)

![Demo > Widgets > Basic](https://i.imgur.com/sgpZARt.gif)

**Standalone, minimal, complete and verifiable example:**
```cpp
ImGui::Begin("Example Bug");
if(ImGui::BeginPopup("popup")) {
  ImGui::Text("Contents");
  ImGui::EndPopup();
}
ImGui::Text("right-click me");
ImGui::OpenPopupOnItemClick("popup", ImGuiPopupFlags_MouseButtonRight);
ImGui::End();
```

**Comment by ocornut** [comment]
Thank you, I could confirm the bug.
With the calls done in this order, the window doesn't have time to get focus and popup gets culled on next frame.

**Comment by ocornut** [comment]
Should be fixed by 3e4c8905
(We're not currently 100% sure this won't have another side effect, I guess we'll find out if it does)
